packages/api/src/controllers/session.ts

```diff
@@ -204,6 +216,26 @@ const adminOnlyFields: (keyof DBSession)[] = ["deleted", "broadcasterHost"];
 
 const privateFields: (keyof DBSession)[] = ["recordObjectStoreId", "version"];
 
+export async function getSourceRecordingBySessionId(
+  sessionId: string,
+  req: Request
+) {
+  const session = await db.session.get(sessionId);
+
+  if (!session) {
+    console.log(`
+      session: recording by session id: no recording found for session ${sessionId}
+    `);
+    return {
+      url: null,
+      session: null,
+      objectStoreId: null,
+    };
+  }
+
+  return await buildRecordingUrl(session, req);
+}
+
```

Comment:
This function seems unnecessary. Every caller of it already has the `session` object available. Then we are doing another round-trip to the DB to fetch it again. Callers can just use `buildRecordingUrl` directly.

Reply:
Yeah, made this in case we later want to include session source recording URL in the playback API endpoint as well, but I can remove it, don't think it's going to happen